finality-aleph/src/data_io.rs

```diff
@@ -117,30 +116,30 @@ where
     block_requester: RB,
     config: DataStoreConfig,
     _phantom: PhantomData<BE>,
-    network: N,
+    messages_from_network: Arc<tokio::sync::Mutex<R>>,
```

Comment:
Nit: Please import Mutex, rather than right in this way

Reply:
There's another `Mutex` name imported in this file (from `parking_lot`) so either I'd need to rename that entry or I can import tokio Mutex as sth like `TokioMutex`, and I think latter is less readable than current version. For the moment I'll leave it as it is then, I understand there's a convention to import and use first-level-names mostly, but on the other hand we do use symbols like `std::fmt::Debug` even in this file.

Reply:
Blah, we really need to refactor this... Also remove the `parking_lot` mutexes, they are dangerous in async code.

Not in this PR ofc, the specific changes require more though.